lib/connection_manager.go

```diff
@@ -700,10 +586,35 @@ func (cmgr *ConnectionManager) addPeer(pp *Peer) {
 	}
 
 	peerList[pp.ID] = pp
+	cmgr.connectedPeers[pp.ID] = pp
+}
+
+func (cmgr *ConnectionManager) SendMessage(msg DeSoMessage, peerId uint64) error {
+	if peer, ok := cmgr.connectedPeers[peerId]; ok {
+		glog.V(1).Infof("SendMessage: Sending message %v to peer %d", msg.GetMsgType().String(), peerId)
+		peer.AddDeSoMessage(msg, false)
+	} else {
+		return fmt.Errorf("SendMessage: Peer with ID %d not found", peerId)
+	}
+	return nil
```

Comment:
```suggestion
	if peer, ok := cmgr.connectedPeers[peerId]; ok {
		glog.V(1).Infof("SendMessage: Sending message %v to peer %d", msg.GetMsgType().String(), peerId)
		peer.AddDeSoMessage(msg, false)
		return nil
	}
	return fmt.Errorf("SendMessage: Peer with ID %d not found", peerId)
```

Reply:
OR
```suggestion
	if peer, ok := cmgr.connectedPeers[peerId]; !ok {
		return fmt.Errorf("SendMessage: Peer with ID %d not found", peerId)
	}
	glog.V(1).Infof("SendMessage: Sending message %v to peer %d", msg.GetMsgType().String(), peerId)
	peer.AddDeSoMessage(msg, false)
	return nil
```

Reply:
Ah yea good observation, I actually simplified it in a later PR. Moved it to this PR.